modules/eks/cluster/modules/node_group_by_az/main.tf

```diff
@@ -0,0 +1,71 @@
+data "aws_subnet_ids" "private" {
+  count = local.enabled ? 1 : 0
+
+  vpc_id = var.cluster_context.vpc_id
+
+  tags = {
+    (var.cluster_context.subnet_type_tag_key) = "private"
+  }
+
+  filter {
+    name   = "availability-zone"
+    values = [var.availability_zone]
+  }
+}
+
+module "az_abbreviation" {
+  source  = "cloudposse/utils/aws"
+  version = "0.8.1"
+}
+
+locals {
+  enabled         = module.this.enabled && length(var.availability_zone) > 0
+  sentinel        = "~~"
+  subnet_ids_test = coalescelist(flatten(data.aws_subnet_ids.private[*].ids), [local.sentinel])
+  subnet_ids      = local.subnet_ids_test[0] == local.sentinel ? null : local.subnet_ids_test
+  az_map          = var.cluster_context.az_abbreviation_type == "short" ? module.az_abbreviation.region_az_alt_code_maps.to_short : module.az_abbreviation.region_az_alt_code_maps.to_fixed
+  az_attribute    = local.az_map[var.availability_zone]
+}
+
+module "eks_node_group" {
+  source  = "cloudposse/eks-node-group/aws"
+  version = "0.27.3"
```

Comment:
`2.6.0` is available. if there are major changes required for that upgrade, ignore this